### ISSUE TITLE
`LeaderElection.Close` is noop if `RelinquishIntervalSpec` is unset

### ISSUE THREAD

**Issue by agis** [drio-ai/leaderelection#3]
Hello. I've noticed that `closeRelinquishJob` does nothing if `LeaderElectionConfig.RelinquishIntervalSpec` is unset, even if `RelinquishInterval` is set: https://github.com/drio-ai/leaderelection/blob/2524d273ac890af11d9c7194aef2f329720df401/election.go#L63-L68

Is this expected? Are we supposed to set only `IntervalSpec` vs `Interval`?

**Edit**: Are we supposed to call `RelinquishLeadership` ourselves or is this taken care by this library? For example, if we receive an error from `Run` and we want to let someone else take over, it seems that just calling `Close` is insufficient.

Thanks in advance.

**Comment by agis** [comment]
@sraghavan-drio any insights on this one (and possibly #1 and #2)? Thanks in advance.

**Comment by sraghavan-drio** [comment]
@agis

Sorry, just got back from my vacation and looking at these issues. Either IntervalSpec or Interval should work. This looks like a bug if it is not working for you. I understand you have RelinquishInterval set but not RelinquishIntervalSpec. In this case you are seeing the leader relinquish leadership? Is this correct? Are you using Postgres or Redis?

**Comment by agis** [comment]
It does seem to work as expected, it's just that the quoted code stood out to me, but I probably haven't understood it. Is `le.cS.Stop()` only relevant to `IntervalSpec`?

**Comment by sraghavan-drio** [comment]
> It does seem to work as expected, it's just that the quoted code stood out to me, but I probably haven't understood it. Is `le.cS.Stop()` only relevant to `IntervalSpec`?

Yes, this is correct. RelinquishIntervalSpec is implemented using https://pkg.go.dev/github.com/robfig/cron/v3. This module starts a goroutine to process jobs. When we are done, this goroutine must be stopped and that is what closeRelinquishJob does. The functions initRelinquishJob, closeRelinquishJob, startRelinquishJob and stopRelinquishJob are all related. They are all noop if RelinquishIntervalSpec is not defined.
